[#168189131] Add info about communication with PagoPA and kubernetes … (#244)
teamdigitale/io-pagopa-proxy: README.md

Also be sure to use a valid configuration for REDIS server or install it on local machine.

> ## Real-world deployments and Kubernetes settings

Reviewer: you mean production deployments?

> In real-world deployments, the application communicates with PagoPA performing a TLS mutual authentication and establishing a TLS encrypted tunnel.

Reviewer: ```suggestion
In production deployments, the application communicates with PagoPA performing a TLS mutual authentication and establishing a TLS encrypted tunnel.
```

Authentication and tunnel mechanisms are transparent for the pagopa-proxy application and are offloaded to third-party components.

> Normally, the pagopa-proxy application is deployed as a set of Kubernetes resources. A specific *helm-chart* has been developed and currently used to deploy the application. The chart includes some configuration files that deploy a sidecar container that manages the authentication and encryption mechanisms with PagoPA.

Reviewer: ```suggestion
Normally, the `pagopa-proxy` application is deployed as a set of Kubernetes resources. A specific *helm-chart* has been developed and currently used to deploy the application. The chart includes some configuration files that deploy a sidecar container that manages the authentication and encryption mechanisms with PagoPA.
```

> More information about how the pagopa-proxy application communicates with PagoPA, and how to deploy the helm-chart can be found in the [pagopa-proxy helm-chart folder in the io-infrastructure-post-config repository](https://github.com/teamdigitale/io-infrastructure-post-config/tree/master/pagopa-proxy).

Reviewer: ```suggestion
More information about how the `pagopa-proxy` application communicates with PagoPA, and how to deploy the helm-chart can be found in the [pagopa-proxy helm-chart folder in the io-infrastructure-post-config repository](https://github.com/teamdigitale/io-infrastructure-post-config/tree/master/pagopa-proxy).
```
